content/blog/argocd-canary.en.md

```diff
@@ -0,0 +1,499 @@
+rolling update and be able to rollback if something wrong happen.
+
+### Our application and its releases
+
+I will use a simple application which provides a webservice with a json output
+like this:
+
+```
+{
+  "color": "red",
+  "status": "ok"
+}
+```
+
+Very simple. We will watch `color` to track application upgrades and `status`
+as a health metric.
+
+Here's three releases of our application:
+
+- `1.0` with `color` at `red`
+- `2.0` with `color` at `blue`
+- `3.0` with `color` at `black` and `status` at `nok`
+
+
+```yaml
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: colorapi
+  labels:
+    app: colorapi
+spec:
+  replicas: 5
+  revisionHistoryLimit: 2
+  selector:
+    matchLabels:
+      app: colorapi
+  template:
+    metadata:
+      labels:
+        app: colorapi
+    spec:
+      containers:
+      - name: colorapi
+        image: particule/simplecolorapi:1.0
+        imagePullPolicy: Always
+        ports:
+        - name: web
+          containerPort: 5000
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: colorapi
+spec:
+  ports:
+  - port: 80
+  selector:
+    app: colorapi
+  type: LoadBalancer
+```
+
+
+### ArgoCD
+
+As suppose by its name, ArgoCD handles the Continuous Delivery part, it allows
+us to deploy specific versions and keep our actual deployments synced with the
+desired state stored in a git repository. ArgoCD supports
+[Helm](https://helm.sh/), [Ksonnet](https://ksonnet.io/),
+[Jsonnet](https://jsonnet.org/), [Kustomize](https://kustomize.io/) and of
+course standalone Kubernetes manifests. We will use the laters for this
+article.
+
+I let you walk through the
+[official "getting started" guide](https://argoproj.github.io/argo-cd/getting_started/)
+to deploy Argo on your Kubernetes cluster. You should also install the ArgoCD
+CLI. Then we can start deploy our first application.
+
+```console
+$ argocd app create colorapi --repo https://github.com/particuleio/demo-concourse-flux.git --path deploy --dest-server https://kubernetes.default.svc --dest-namespace default
+$ argocd app sync colorapi
+$ kubectl get pod
+NAME                          READY   STATUS    RESTARTS   AGE
+colorapi-7ccb9d965b-8pr54   1/1     Running   0          22s
+colorapi-7ccb9d965b-9vtmp   1/1     Running   0          22s
+colorapi-7ccb9d965b-c7jkf   1/1     Running   0          22s
+colorapi-7ccb9d965b-klt7q   1/1     Running   0          22s
+colorapi-7ccb9d965b-s2ftc   1/1     Running   0          22s
+```
+
+Our first test will be a modification of our replicas count. We are going to
+change that parameter in our Deployment, commit/push our code and see if ArgoCD
+would update our actual Deployment. ArgoCD can track either a tag, a commit or
+a branch. If you track a git tag, don't forget to update it to the latest
+commit.
+
+<https://argoproj.github.io/argo-cd/user-guide/tracking_strategies/>
+
+A poll occurs every 3 minutes, you should't wait long before the scheduling of
```

Comment:
it **shouldn't** / **should not**  take long before our new replicas are scheduled